website/src/views/PageStateHandler.ts

```diff
@@ -305,3 +282,162 @@ export class GenericCompareVariantsStateHandler extends CompareVariantsStateHand
+            setSearchFromDateRange(search, this.constants.mainDateField, pageState.baselineFilter.dateRange);
+        }
+
+        return `${this.pathname}?${search}`;
+    }
+
+    public baselineFilterToLapisFilter(baselineFilter: BaselineFilter): LapisFilter {
+        return toLapisFilterWithoutVariant({ baselineFilter }, this.constants);
+    }
+
+    public variantFiltersToNamedLapisFilters(pageState: CompareVariantsData): NamedLapisFilter[] {
+        const baselineFilter = this.baselineFilterToLapisFilter(pageState.baselineFilter);
+
+        return Array.from(pageState.variants.values()).map((variantFilter) => {
+            return {
+                lapisFilter: {
+                    ...baselineFilter,
+                    ...variantFilter.lineages,
+                    ...variantFilter.mutations,
+                },
+                displayName: this.toDisplayName(variantFilter),
+            };
+        });
+    }
+
+    public getEmptyVariantFilterConfig(): VariantFilterConfig {
+        return this.toLineageAndMutationFilterConfig({
+            lineages: {},
+            mutations: {},
+        });
+    }
+
+    public toVariantFilterConfigs(pageState: CompareVariantsData): Map<Id, VariantFilterConfig> {
+        return new Map<Id, VariantFilterConfig>(
+            Array.from(pageState.variants, ([key, variant]) => [key, this.toLineageAndMutationFilterConfig(variant)]),
+        );
+    }
+
+    private toDisplayName(variantFilter: VariantFilter) {
+        return [
+            ...Object.values(variantFilter.lineages),
+            ...(variantFilter.mutations.nucleotideMutations ?? []),
+            ...(variantFilter.mutations.aminoAcidMutations ?? []),
+            ...(variantFilter.mutations.nucleotideInsertions ?? []),
+            ...(variantFilter.mutations.aminoAcidInsertions ?? []),
+        ].join(' + ');
+    }
+
+    private writeVariantsDataToSearchParams(searchOfFilter: URLSearchParams, filter: VariantFilter): void {
+        setSearchFromLapisVariantQuery(searchOfFilter, filter, getLineageFilterFields(this.constants.lineageFilters));
+    }
+
+    private getFilter(filterParams: Map<string, string>): VariantFilter {
+        return {
+            ...getLapisVariantQuery(filterParams, getLineageFilterFields(this.constants.lineageFilters)),
+        };
+    }
+
+    private toLineageAndMutationFilterConfig(variant: VariantFilter) {
+        return {
+            lineageFilterConfigs: getLineageFilterConfigs(this.constants.lineageFilters, variant.lineages),
+            mutationFilterConfig: variant.mutations,
+        };
+    }
+}
+
+function decodeFiltersFromSearch(search: URLSearchParams, handleNonMatchingKey: (key: string) => void = () => {}) {
+    const filterMap = new Map<Id, Map<string, string>>();
+
+    for (const [key, value] of search) {
+        const keySplit = key.split($);
+        if (keySplit.length !== 2) {
+            handleNonMatchingKey(key);
```

Comment:
Do we actually need that difference? Or could we throw in any case (or ignore the error)? (I think I'd prefer throwing, but not sure)

Reply:
I now removed the error.

Reply:
Ah I see. On the compare side-by-side page, all keys had to be split. I'm not sure whether it's better to silently ignore unknown parameters or throwing an error. Let's see...